Hashing/C++/9_Distinct Echo Substrings (HASHING) .cpp

```diff
@@ -0,0 +1,48 @@
+//brute force approach
+//time complexity=0(n2)
+//space complexity 0(n)
+
+class Solution {
+public:
+    int distinctEchoSubstrings(string text) {
+        int n=text.length();
+        if(n==1) return 0;
+        if(n==2){
+            if(text[0]==text[1]) return true;//considering the base cases one string can be formed in such case
+            return false;
+        }
+        set<string_view>s;//stored in a set to be in order
+        string_view str=text;
+        for(int i=0;i<n;i++){//iterating throgufh the string 
+           for(int len=2;i+len-1<n;len+=2){//trying to find the substrings
+               string_view a1=str.substr(i,len/2);
+               string_view a2=str.substr(i+len/2,len/2);//string view is used for some string type interface is used 
+               if(a1==a2){//sice its a part of the string concatenated to form one 
+                   
+                   s.insert(a1);.//inserting all such strings that can be formed
+               }
+                   
+           }
+        }
+        return s.size();//returning
+    }
+};
+int main(){
+string text;
+cin>>text;
+distinctEchoSubstrings(string text);
```

Comment:
You are returning s.size() in the function. You just need to print distinctEchoSubstrings(string text) not s.size()
@soma2000-lang